### ISSUE TITLE
Switch to Embedded Package Icons

### ISSUE THREAD

**Issue by wtgodbe** [dotnet/corefx#41371]
Once we move to the 3.0.100 SDK in our rel/3.0 branches, we'll start hitting NU5048. This is about moving from `PackageIconUrl` (a link in the nuspec to a URL where the package icon image lives), to an embedded image in the package itself. We need to react to this in CoreFx once we update the SDK - see https://github.com/aspnet/Blazor/pull/1896 for an example PR.

CC @ericstj @Anipik @safern @ViktorHofer 

We should make sure somebody is doing this for CoreClr/Core-Setup/Standard as well

**Comment by ViktorHofer** [comment]
I'm surprised that we didn't hit `NU5048` yet. We switched to 3.0.100 in master already and have a PR up that update to 3.0.100 in the release/3.0.

**Comment by wtgodbe** [comment]
I don't even know how CoreFx/Core-Setup get IconUrl's into their .nuspec's - I confirmed that the packages we shipped with GA do have the `iconUrl` metadata in their .nuspec's, but there is no instance of the `PackageIconUrl` property in CoreFx. Maybe we're using some Arcade/SDK default?

**Comment by ericstj** [comment]
We do it here: https://github.com/dotnet/arcade/blob/0f1befce6666a71ae459ac61507a00535bbdfe75/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.targets#L51.

Work would be to add a new property and expose it on GenerateNuSpec.  We could bundle the default icon with the packaging infra, same as we do the other files and make it happen for our stuff automatically.

Perhaps NU5048 isn't triggered in PackageBuilder?  I'd have to look at where they inserted that check.

**Comment by ericstj** [comment]
Yeah, NuGet only runs this during the Pack task.  Maybe we should consider adding that to our own pack task (so long as we still need it and cannot move to using the nuget pack task).

**Comment by danmosemsft** [comment]
@ViktorHofer @wtgodbe can you clarify the importance of this one? It's marked 3.0 but nobody is working on it.

Is this required to ship 3.0 (or 3.1)?

> We should make sure somebody is doing this for CoreClr/Core-Setup/Standard as well

Is this confirmed, can you link issues?

**Comment by ericstj** [comment]
I have a fix for this.   I'll be putting it into arcade, we just need to make sure we get it to all the right places.

**Comment by ericstj** [comment]
@wtgodbe @mmitche this change is in to arcade, do you guys plan to manually flow Arcade into CoreFx release branches?

**Comment by mmitche** [comment]
@ericstj Ya, once we finish 3.0.1

**Comment by ericstj** [comment]
Assume the same will happen in 3.1?

**Comment by mmitche** [comment]
Yeah, those arcade subscriptions flow to both places

**Comment by ericstj** [comment]
Hey folks, just checked release/3.1 and this didn't happen.  @wtgodbe don't we want this in the preview 3 build?
